src/main.zig

```diff
@@ -2005,6 +2014,16 @@ fn buildOutputType(
         else => false,
     };
 
+    // Always output import libraries (.lib) when building for msvc to replicate
+    // `link` behavior. lld does not always output import libraries so on the
+    // gnu abi users must set out_implib.
+    if (output_mode == .Lib and emit_bin == .yes and target_info.target.abi == .msvc and out_implib == null) {
```

Comment:
If `emit_bin` is set when building static archives, this check will let that through, so you should also check for dynamic linking `comp.bin_file.options.link_mode == .Dynamic`.

Reply:
My understand is the import library is only needed when building dynamic libraries. When building static libraries the import parts are built into the archive in all platforms, and this is why they share the extension in windows.

Since i dont have windows handy to test this is simply my assumption and some googling suggests its probably not wrong.

Reply:
Your understanding is correct and the way we signal a dynamic library in the linker is with a combination `comp.bin_file.options.output_mode.Lib` and `comp.bin_file.options.link_mode.Dynamic`, and I'm pointing at the fact that we should check for this combo and not only if it's a `.Lib`.

Reply:
Sorry and that .Lib is only set when building shared libs, from line 1510

Reply:
That's not correct. We always discriminate shared objects from static archives using the combo. See [here](https://github.com/ziglang/zig/blob/eb010ce65ddddb23ceddae86d377432cf1b2b01c/src/link/MachO.zig#L628) for instance.

Reply:
I see, I think the issue is I am adding this outside of the .cc/.cpp args check. Meaning it will affect other build modes. Maybe this is what you intend with your comment on aligning build-lib? I can add the dynamic check as well but should we restrict this to cc/cpp modes as well until the option can be added as a regular zig flag instead of strictly a linker flag?

Reply:
I think we should be fine with other build modes if we set the default for the flag in further down the stack in `Compilation` rather than here. Anyhow, my suggestion would be to try it out and if things break along the way, then we can figure out what to fix and where. If you don't feel like wanting to pursue this, I'm happy to take over from here. Otherwise, I'm happy to work with you to get this merged.

Reply:
Sure, by all means if you have time to work on this id be happy if you could work on it as Im mostly just muddling my way through zig to get something that works. Of course feel free to use anything in this pr as you see fit.